src/algebra/big_operators/basic.lean

```diff
@@ -1332,6 +1332,24 @@ begin
   rwa eq_of_mem_of_not_mem_erase hx hnx
 end
 
+@[to_additive]
+lemma ite_exists_eq_prod {f : α → Prop} [Π i, decidable (f i)]
+  (hf : (s : set α).pairwise_disjoint f) (a : β) :
+  ite (∃ i ∈ s, f i) a 1 = ∏ i in s, ite (f i) a 1 :=
+begin
+  classical,
```

Comment:
Can't you get here faster with split_ifs and`finset.sum_eq_single`?

Reply:
Well spotted, thanks!